src/ui/NhekoGlobalObject.cpp

```diff
@@ -190,3 +190,63 @@ Nheko::setWindowRole([[maybe_unused]] QWindow *win, [[maybe_unused]] QString new
     QXcbWindowFunctions::setWmWindowRole(win, newRole.toUtf8());
 #endif
 }
+
+QString
+Nheko::getCommandFromText(const QString &text)
+{
+    if (text.startsWith('/')) {
+        int command_end = text.indexOf(QRegularExpression(QStringLiteral("\\s")));
+        if (command_end == -1)
+            command_end = text.size();
+        auto command = text.mid(1, command_end - 1);
+        if (command.isEmpty() || command == QLatin1String("/"))
+            return {};
+        else {
+            return command;
+        }
+    } else
+        return {};
+}
+
+bool
+Nheko::isInvalidCommand(QString command) const
+{
+    if (command.size() <= 0)
+        return false;
+
+    static const QStringList validCommands{QStringLiteral("/me"),
```

Comment:
I looked at the command completer, but I decided that it was going to be the wrong approach as-is.

> The bigger concern is, that this probably should be a property of the InputBar instead of the UI layer. It could just be a string property or so, if you want easy access to the actual command.

As in expose a property from C++ that tells QML if the warning should be displayed?

Reply:
Yes,

Reply:
Done